### PR TITLE
ImmutableResultSet unittests

### LINKED ISSUE #229: Expand and fix unit tests for ImmutableResultSets
Since a few Pull Requests, unit tests fail again.

Line 176 fails:
https://github.com/VForWaTer/metacatalog/blob/995c52224a20c165a9feb48c900cadcb99216668/metacatalog/test/test_api_add_find.py#L173-L176

`dummy_results` is a list of two ImmutableResultSets here.

`__init__()` does not accept an instance of type `ImmutableResultSet` anymore and we can delete all lines of code, where `isinstance(instance, ImmutableResultSet)` is checked:
https://github.com/VForWaTer/metacatalog/blob/995c52224a20c165a9feb48c900cadcb99216668/metacatalog/util/results.py#L64
https://github.com/VForWaTer/metacatalog/blob/995c52224a20c165a9feb48c900cadcb99216668/metacatalog/util/results.py#L84-L86

We also need additional tests for `ImmutableResultSets`:
- [x] Test that the `ImmutableResultSet` of an `ImmutableResultSet` is the same `ImmutableResultSet` via its checksum
- [x] Test that `ImmutableResultSets` are never empty.
- [x] Test `.to_dict()` method

### PR DESCRIPTION
**Pull request by AlexDo1**
This PR adds further unittests for `ImmutableResultSets`.  
Closes #229

### CODE REVIEW

**Review by mmaelicke — CHANGES_REQUESTED** [review]

File: metacatalog/test/test_result_set.py [hunk 1]
```diff
@@ -85,6 +85,57 @@ def add_data(session):
...
+
+
+def result_set_check_to_dict(session):
+    """
+    Check the `to_dict()` method of ImmutableResultSet
+    """
+    # list of expected keys in ImmutableResultSet dictionary
+    expected_keys = ['author', 'title', 'details', 'lastUpdate', 'keywords', 'id', 'license', 'uuid', 
+                     'variable', 'authors', 'locationShape', 'embargo_end', 'publication', 'abstract', 
+                     'version', 'location', 'embargo', 'isPartial']
```
[the remark this answers is not in the document]
Comment (AlexDo1) [reply metacatalog/test/test_result_set.py:129]: I shortened the list of expected keys as you suggested.

File: metacatalog/test/test_result_set.py [hunk 2]
```diff
@@ -140,7 +191,7 @@ def result_list_check_append(session):
     
     return True
 
-@pytest.mark.depends(on=['add_find'], name='resutlset')
+@pytest.mark.depends(on=['add_find'], name='resultset')
```
[the remark this answers is not in the document]
Comment (AlexDo1) [reply metacatalog/test/test_result_set.py:194]: Yes, until now, no other test depends on the resultset test.

File: metacatalog/test/test_result_set.py [hunk 3]
```diff
@@ -153,3 +204,6 @@ def test_result_set():
     assert add_entries(session)
     assert add_data(session)
     assert result_list_check_temporal_scale(session)
+    assert result_set_check_resultset_of_resultset(session)
+    assert result_set_check_resultset_of_resultset(session)
```
[the remark this answers is not in the document]
Comment (AlexDo1) [reply metacatalog/test/test_result_set.py:208]: Yes, thanks!

File: metacatalog/test/test_result_set.py [hunk 4]
```diff
@@ -85,6 +85,57 @@ def add_data(session):
...
+    entry = api.find_entry(session, title="3-dimensional windspeed data")[0]
+
+    # assert if entry is associated to any groups
+    assert entry.associated_groups == []
+
+    # ImmutableResultSet of entry
+    rs = ImmutableResultSet(entry)
+
+    # assert if ImmutableResultSet has no members and if title of contained entry is wrong
+    assert rs._members > 0
```
[the remark this answers is not in the document]
Comment (AlexDo1) [reply metacatalog/test/test_result_set.py:116]: Yes, I adapted that.

**Review by mmaelicke — APPROVED** [review]

**Pull request merged**